internal/provider/data_file.go

```diff
@@ -0,0 +1,146 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/neuspaces/terraform-provider-system/internal/client"
+	"github.com/neuspaces/terraform-provider-system/internal/validate"
+	"path"
+)
+
+const dataFileName = "system_file"
+
+func dataFile() *schema.Resource {
+	return &schema.Resource{
+		Description: fmt.Sprintf("`%s` retrieves information about a file on the remote system.", dataFileName),
+
+		ReadContext: dataFileRead,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			resourceFileAttrId: {
+				Description: "ID of the file",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			resourceFileAttrPath: {
+				Description:      "Path to the file. Must be an absolute path.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.AbsolutePath(),
+			},
+			resourceFileAttrMode: {
+				Description:      "Permissions of the file in octal format like `755`. Defaults to the umask of the system.",
+				Type:             schema.TypeString,
+				Computed:         true,
+				ValidateDiagFunc: validate.FileMode(),
+			},
+			resourceFileAttrUser: {
+				Description: "Name of the user who owns the file",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			resourceFileAttrUid: {
+				Description: "ID of the user who owns the file",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			resourceFileAttrGroup: {
+				Description: "Name of the group that owns the file",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			resourceFileAttrGid: {
+				Description: "ID of the group that owns the file",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			resourceFileAttrContent: {
+				Description: fmt.Sprintf("Content of the file. Only recommended for small text-based payloads such as configuration files etc. In a terraform plan,  The content will be stored in plain-text in the terraform state. Mutually exclusive with attributes `%[2]s` and `%[3]s`.", resourceFileAttrContent, resourceFileAttrContentSensitive, resourceFileAttrSource),
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   false,
+			},
+			resourceFileAttrContentSensitive: {
+				Description: fmt.Sprintf("Content of the file similar to `%[1]s` attribute but with enabled sensitive flag. Prefer `%[2]s` to `%[1]s` to avoid leak of the content in the terraform log output. Mutually exclusive with attributes `%[1]s` and `%[3]s`.", resourceFileAttrContent, resourceFileAttrContentSensitive, resourceFileAttrSource),
```

Comment:
Please review the descriptions and if they still apply to the data source. Here, attributes are mentioned which do not exist in the data source.

Reply:
ok, removed by removing implementation of sensitive content